### PR TITLE
Stripe Hooks and flushed out some specs

### PR DESCRIPTION
**Pull request by barberj**
https://assembly.com/signupsumo/bounties/69

Update customer JSON payload so that `subscription.active?` is true when stripe posts to hook endpoint.
`subscription.active?` checks the customer JSON for `current_period_end` to verify the date is less then now.

~~Simplified bounty ignoring customer deletes since the `current_period_end` would not move forward and have same effect.~~

### CODE REVIEW

**Comment by barberj** [comment]
@vanstee review and deploy please?

Do you have access to stripe? You will need to login to the stripe and add the webhook.

screen shot 2015-02-20 at 10 08 22 pm

url should be "https://signupsumo.com/subscriptions/webhooks"
We only need the payment_succeed event since a failed payment would be no different then the period expiring. Lets just keep it simple and not worry bout all the conditional paths.

screen shot 2015-02-20 at 10 09 06 pm

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -2,16 +2,20 @@
...
+
+Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
 
 ActiveRecord::Migration.maintain_test_schema!
 
 RSpec.configure do |config|
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
 
-  config.use_transactional_fixtures = true
+  config.use_transactional_fixtures = false
```
Comment (barberj) [inline spec/rails_helper.rb]: this was screwing up the postgres json query for some reason... http://stackoverflow.com/questions/21138207/activerecordstatementinvalid-pg-infailedsqltransaction

**Comment by barberj** [comment]
@asm-products/signupsumo-web any one up for a review? Just getting started on the core here so not sure policy on merging your own work. Also I don't have the stripe or deployment access (that i know of). Is that a core thing or something assembly handles? 

Thanks

**Comment by barberj** [comment]
screen shot 2015-02-21 at 8 26 27 pm

File: app/controllers/subscriptions_controller.rb [hunk 2]
```diff
@@ -19,6 +20,23 @@ def create
     end
   end
 
+  def stripe_hook
+    if customer_id = params.try(:[], 'data').try(:[], 'object').try(:[], 'customer')
+      if subscription = Subscription.where("customer->>'id' = ?", customer_id).first
```
Comment (vanstee) [inline app/controllers/subscriptions_controller.rb]: Fancy :) We might want to index this eventually, but looks good for now.

**Comment by vanstee** [comment]
The code looks good. The only thing that probably needs a little more work is the notification email. Right now it probably too closely resembles a regular influencer notification. I'll merge this and create a bounty to revise the email.

**Comment by vanstee** [comment]
@barberj Is there a bounty to award for this one? If not feel free to create one with a link to this PR and award it to yourself.

**Comment by barberj** [comment]
i awarded myself and tipped @kenrick since I was unaware how to split a bounty. Tip amount may be inadequate

**Comment by vanstee** [comment]
Sweet ok. That's alright. We're working on a better way to handle situations like that. Tipping away part of your award or revaluing the bounty and reawarding are the two ways I've been handling stuff like that lately.

**Pull request merged**